### PR TITLE
FEAT: RBAC: Read anonymous user from DB

### PR DESCRIPTION
**Pull request by sidhantgoel**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description

Currently anonymousUser is static and hence single instance of it has been created in code itself and there are no properties associated with it. With introduction of rbac anonymousUser will be created for every tenant in the database like a normal user and hence will have properties like a normal user.

With this change anonymousUser will be fetched from DB and cached for later access. `anonymous().principal()` is replaced with `anonymous().authenticationFilter()` to allow for dynamically setting of anonymousUser based on tenant, currently it is not using tenant but loading 'anonymousUser' from DB as currently there exists only single default tenant.

TODO will be to figure out tenantId from incoming request and load corresponding 'anonymousUser' when we move to multi tenancy.

> Please include a summary of the changes and which issue has been fixed. Please also include relevant motivation
> and context. List any dependencies that are required for this change.

Fixes # (issue)

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                .map(user ->  new AnonymousAuthenticationToken(this.key, user, this.authorities));
+    }
+
+    @Override
+    public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
+		return ReactiveSecurityContextHolder.getContext().switchIfEmpty(Mono.defer(() -> {
+            return createAuthenticationMono(exchange)
+                    .flatMap(authentication -> {
+                        SecurityContext securityContext = new SecurityContextImpl(authentication);
+                        log.debug("Populated SecurityContext with anonymous token: '{}'", authentication);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java:58]: Are these logs necessary? If not, then can we please remove them?

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java (hunk 1 above)
Comment (sidhantgoel) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java:58]:
```suggestion
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		return ReactiveSecurityContextHolder.getContext().switchIfEmpty(Mono.defer(() -> {
+            return createAuthenticationMono(exchange)
+                    .flatMap(authentication -> {
+                        SecurityContext securityContext = new SecurityContextImpl(authentication);
+                        return chain.filter(exchange)
+                                .contextWrite(ReactiveSecurityContextHolder.withSecurityContext(Mono.just(securityContext)))
+                                .then(Mono.empty());
+                    });
+		})).flatMap((securityContext) -> {
+            log.debug("SecurityContext contains anonymous token: '{}'", securityContext.getAuthentication());
```
Comment (sidhantgoel) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/filters/AppsmithAnonymousAuthenticationWebFilter.java:63]:
```suggestion
```

**Review by trishaanand — APPROVED** [review]

**Pull request closed without merging**